lib/log_download.py

```diff
@@ -0,0 +1,138 @@
+import datetime
+from dateutil.rrule import rrule, DAILY
+from datetime import timedelta
+from BeautifulSoup import BeautifulSoup
+from logging.config import fileConfig
+import urllib2 as ulib
+import logging
+import re
+import time
+import os
+
+# load the logging configuration
+fileConfig('logging_config.ini')
+
+# Global variables
+UBUNTU_ENDPOINT = "https://irclogs.ubuntu.com/{}/{month:02d}/{day:02d}"
+UBUNTU_CHANNEL_ENDPOINT = "https://irclogs.ubuntu.com/{}/{month:02d}/{day:02d}/%23{channel}"
+SCUMMVM_ENDPOINT = "http://www.enderboi.com/log.php?log=scummvm.log.{day:02d}{month}{year}&format=text"        
+    
+def date_separate(dt):
+    """
+    Args:
+        dt (date object) : A valid date object
+    
+    Returns:
+        day  (int) : day corresponding to input date object
+        month(int) : month corresponding to input date object
+        year (int) : year corresponding to input date object
+    """
+    return dt.day, dt.month, dt.year
+
+def save_log(log, save_dir, dt, name):
+    """
+    Args:
+        log      (str) : chat log as text
+        save_dir (str) : location of directory to save the log
+        dt       (date): date corresponding to the log
+        name     (str) : name of the file to be saved
+    """
+    
+    if not (os.path.exists(save_dir) and os.path.isdir(save_dir)):
+        logging.info("Save Directory does exist. Creating a new directory {}".format(save_dir))
+        os.makedirs(save_dir)
+    
+    day,month,year = date_separate(dt)
```

Comment:
perhaps, this can be done using strftime; If we can do so, then why have a separate function?